discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -58,9 +59,15 @@ def save_time(self, labels=None, start_time=None):
 
     def save(self, value, labels=None):
         if labels:
-            self.metric.labels(**labels).observe(value)
+            if self.metric_type == PrometheusType.HISTOGRAM:
```

Comment:
not necessary in this PR but with more complexity added to this file we should add some docstrings to each of these functions